frontend/src/components/Dashboard.vue

```diff
@@ -0,0 +1,321 @@
+<script setup lang="ts">
+import Dropdown, { type DropdownChangeEvent } from 'primevue/dropdown';
+import ConfirmDialog from 'primevue/confirmdialog';
+import Button from '../components/common/Button.vue';
+import InputText from 'primevue/inputtext';
+import { useConfirm } from 'primevue/useconfirm';
+import { ApiServiceFactory } from '@/services/ApiServiceFactory';
+import { computed, onMounted, ref, reactive } from 'vue';
+import {
+    applicationsUserAdministers,
+    isApplicationSelected,
+    selectedApplication,
+    setSelectedApplication,
+    selectedApplicationDisplayText,
+    useNotificationMessage,
+} from '@/store/ApplicationState';
+import router from '@/router';
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+import { FilterMatchMode } from 'primevue/api';
+import type { FamApplicationUserRoleAssignmentGet } from 'fam-api/dist/model/fam-application-user-role-assignment-get';
+
+const confirm = useConfirm();
+const apiServiceFactory = new ApiServiceFactory();
+const applicationsApi = apiServiceFactory.getApplicationApi();
+const userRoleAssignmentApi = apiServiceFactory.getUserRoleAssignmentApi();
+const loading = ref<boolean>(false);
+const filters = ref({
+    global: { value: null, matchMode: FilterMatchMode.CONTAINS },
+    'user.user_name': { value: null, matchMode: FilterMatchMode.STARTS_WITH },
+    'role.parent_role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.role_name': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+    'role.client_number.forest_client_number': {
+        value: null,
+        matchMode: FilterMatchMode.CONTAINS,
+    },
+});
+const userRoleAssignments = ref<FamApplicationUserRoleAssignmentGet[]>();
+
+const confirmRemoveMessage = reactive({
+    userName: '',
+    role: '',
+});
+
+onMounted(async () => {
+    // Reload list each time we navigate to this page to avoid forcing user to refresh if their access changes.
+    try {
+        loading.value = true;
+        applicationsUserAdministers.value = (
+            await applicationsApi.getApplications()
+        ).data;
+    } catch (error: any) {
+        return Promise.reject(error);
+    } finally {
+        loading.value = false;
+    }
+
+    if (isApplicationSelected) {
+        getAccessList();
+    }
+});
+const filteredOptions = computed(() => {
```

Comment:
Let's rename it to the selectApplicationFilteredOptions, it confused me a bit if it's for the dropdown or for the data table